indico/modules/events/papers/client/js/components/RevisionInfo.jsx

```diff
@@ -0,0 +1,90 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import paperFileDownloadURL from 'indico-url:papers.download_file';
+
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {useSelector} from 'react-redux';
+import moment from 'moment';
+
+import {Param, Translate} from 'indico/react/i18n';
+import {serializeDate} from 'indico/utils/date';
+
+import UserAvatar from './UserAvatar';
+import {getPaperEvent, getPaperContribution} from '../selectors';
+
+export default function RevisionInfo({revision}) {
+  const {submitter, isLastRevision, id, number, submittedDt, files} = revision;
+  const submitterName = submitter.isSystem ? Translate.string('A User') : submitter.fullName;
+  const contribution = useSelector(getPaperContribution);
+  const event = useSelector(getPaperEvent);
+
+  return (
+    <div className="i-timeline">
+      <div className="i-timeline-item">
+        <UserAvatar user={submitter} />
+        <div
+          className={`i-timeline-item-box header-indicator-left ${
+            !isLastRevision ? 'header-only' : ''
+          }`}
+          id={`revision-info-${id}`}
+        >
+          <div className="i-box-header flexrow">
+            <div className="f-self-stretch">
+              <Translate>
+                <Param name="submitterName" value={submitterName} wrapper={<strong />} /> submitted
+                paper revision <Param name="revisionNumber" value={`#${number}`} />{' '}
+              </Translate>
+              <time dateTime={serializeDate(submittedDt, moment.HTML5_FMT.DATETIME_LOCAL)}>
+                {serializeDate(submittedDt, 'LL')}
+              </time>
+            </div>
+            {!isLastRevision && (
+              <a
+                href="#"
+                className="revision-info-link i-link"
+                data-toggle={`#revision-timeline-${id}, #revision-info-${id} .submission-info`}
+                data-toggle-class={`{"target": "#revision-info-${id}", "class": "header-only"}`}
```

Comment:
shouldn't we use react state for this stuff?

Reply:
Yup, we should ofc. But I didn't think it was that important for now.